src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,102 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int ORIGINAL_SIZE = 10;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[ORIGINAL_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (elements.length == size) {
```

Comment:
Create a separated method(For example,  checkSize() ) for this if and replace it in all places.